### PR TITLE
Rails 4.x porting

### PR DESCRIPTION
**Pull request by dirkmueller**

### CODE REVIEW

**Comment by tboerger** [comment]
LGTM

File: releases/development/master/extra/install-chef-suse.sh [hunk 1]
```diff
@@ -626,13 +627,16 @@ if [ -n "$CROWBAR_FROM_GIT" ]; then
...
     # also need these (crowbar dependencies):
-    zypper -n in rubygem-app_config rubygem-cstruct rubygem-kwalify rubygem-ruby-shadow \
-            rubygem-sass rubygem-i18n sleshammer tcpdump rubygem-sprockets-helpers \
-            rubygem-sprockets-sass rubygem-sqlite3
+    zypper -n in \
+        ruby2.1-rubygem-app_config ruby2.1-rubygem-cstruct ruby2.1-rubygem-kwalify \
+        ruby2.1-rubygem-ruby-shadow ruby2.1-rubygem-sass-3_2 ruby2.1-rubygem-i18n \
+        ruby2.1-rubygem-sprockets-helpers ruby2.1-rubygem-cstruct ruby2.1-rubygem-kwalify \
+        ruby2.1-rubygem-ruby-shadow ruby2.1-rubygem-sprockets-2_11 ruby2.1-rubygem-sqlite3 \
+        sleshammer tcpdump
```
Comment (vuntz) [inline releases/development/master/extra/install-chef-suse.sh]: Did you check these are the right packages to install after the rails4 port? That's the reason I didn't do anything in https://github.com/crowbar/crowbar/pull/2075 ;-)

(Btw, kwalify is there twice).

File: releases/development/master/extra/install-chef-suse.sh [hunk 2]
```diff
@@ -626,13 +627,21 @@ if [ -n "$CROWBAR_FROM_GIT" ]; then
...
-            couchdb rubygem-activesupport
+    zypper -n --gpg-auto-import-keys in ruby2.1-rubygem-chef-server ruby2.1-rubygem-chef \
+        rabbitmq-server couchdb ruby2.1-rubygem-activesupport
 
     # also need these (crowbar dependencies):
-    zypper -n in rubygem-app_config rubygem-cstruct rubygem-kwalify rubygem-ruby-shadow \
-            rubygem-sass rubygem-i18n sleshammer tcpdump rubygem-sprockets-helpers \
-            rubygem-sprockets-sass rubygem-sqlite3
+    zypper -n in \
+        ruby2.1-rubygem-app_config \
```
Comment (tboerger) [inline releases/development/master/extra/install-chef-suse.sh]: This is not used anymore.

**Comment by tboerger** [comment]
If you have synchronized the gems with the Gemfile i'm fine with that

**Comment by tboerger** [comment]
@dirkmueller are the gems in sync with the Gemfile?

**Comment by dirkmueller** [comment]
@tboerger the idea of the install_from_git is to only install those gems from packages that have significant non-upstream patches (e.g. it only needs to install chef and related dependencies). the rest comes from bundler

**Comment by tboerger** [comment]
@dirkmueller in that case this pull request can get closed as this gem gets installed through bundler as well.

File: releases/development/master/extra/install-chef-suse.sh [hunk 3]
```diff
@@ -325,7 +325,7 @@ fi
 if [ -n "$CROWBAR_FROM_GIT" ]; then
     REPOS_SKIP_CHECKS+=" SLES11-SP3-Pool SLES11-SP3-Updates SUSE-Cloud-5-Pool SUSE-Cloud-5-Updates SLES12-Pool SLES12-Updates SLE-12-Cloud-Compute5-Pool SLE-12-Cloud-Compute5-Updates"
 
-    zypper -n in rubygems rubygem-json createrepo
+    zypper -n in ruby2.1-rubygem-json-1_7 createrepo
```
Comment (tboerger) [inline releases/development/master/extra/install-chef-suse.sh]: We dont need the json gem if we are using bundler.

File: releases/development/master/extra/install-chef-suse.sh [hunk 4]
```diff
@@ -679,13 +679,20 @@ if [ -n "$CROWBAR_FROM_GIT" ]; then
...
-            couchdb rubygem-activesupport
+    zypper -n --gpg-auto-import-keys in ruby2.1-rubygem-chef-server ruby2.1-rubygem-chef \
+        rabbitmq-server couchdb ruby2.1-rubygem-activesupport
 
     # also need these (crowbar dependencies):
-    zypper -n in rubygem-app_config rubygem-cstruct rubygem-kwalify rubygem-ruby-shadow \
-            rubygem-sass rubygem-i18n sleshammer tcpdump rubygem-sprockets-helpers \
-            rubygem-sprockets-sass rubygem-sqlite3
+    zypper -n in \
+        ruby2.1-rubygem-cstruct \
```
Comment (tboerger) [inline releases/development/master/extra/install-chef-suse.sh]: In these ruby gem packages are no patches so we dont need to install it manually?

**Comment by dirkmueller** [comment]
@tboerger Again, the point is to install those gems from packages that have significant non-upstream patches.

**Comment by tboerger** [comment]
@dirkmueller You already have written that. And nearly all gems don't have any patches. I think chef is the only gem. So we don't need to install all of the through packages.

**Comment by bmwiedemann** [comment]
:+1:

**Comment by bkutil** [comment]
+1

**Comment by rsalevsky** [comment]
I stumbled again about this PR locks mostly good to me. According to you argumentation you can drop the following packages because they have no significant non-upstream patches.
```
ruby2.1-rubygem-cstruct
ruby2.1-rubygem-i18n
ruby2.1-rubygem-kwalify
ruby2.1-rubygem-ruby-shadow
ruby2.1-rubygem-sass-3_2
ruby2.1-rubygem-sass-rails
ruby2.1-rubygem-sprockets-2_11
ruby2.1-rubygem-sqlite3
```

**Comment by dirkmueller** [comment]
@rsalevsky weird, I thought I removed that already. should be fixed now.

**Comment by rsalevsky** [comment]
+1

**Comment by vuntz** [comment]
+1

**Pull request merged**